src/map/skill.c

```diff
@@ -5856,19 +5859,23 @@ static int skill_castend_id(int tid, int64 tick, int id, intptr_t data)
 		if (target && target->m == src->m) {
 			//Move character to target anyway.
 			int dir, x, y;
+			int dist = 3; // number of cells that asura caster will walk
+
 			dir = map->calc_dir(src,target->x,target->y);
-			if( dir > 0 && dir < 4) x = -2;
-			else if( dir > 4 ) x = 2;
+			if (dir > 0 && dir < 4) x = -dist;
+			else if (dir > 4) x = dist;
 			else x = 0;
-			if( dir > 2 && dir < 6 ) y = -2;
-			else if( dir == 7 || dir < 2 ) y = 2;
+			if (dir > 2 && dir < 6) y = -dist;
+			else if (dir == 7 || dir < 2) y = dist;
 			else y = 0;
-			if (unit->movepos(src, src->x+x, src->y+y, 1, 1)) {
+
+			if (unit->movepos(src, src->x + x, src->y + y, 1, 1)) {
```

Comment:
`unit->movepos(src, src->x + x, src->y + y, 1, 1) == 1`
or
`unit->movepos(src, src->x + x, src->y + y, 1, 1) != 0`

Reply:
Alright. Please NEVER use `== 1` when correcting a condition that mistreats an int as a boolean, ~~UNLESS you're 100% sure that it will never be value besides 1, and even then~~ it's a bad idea, because custom code might expect this condition here to still work the same way it was before.

This should be `!= 0`.

Reply:
Opened an issue about this matter https://github.com/HerculesWS/Hercules/issues/2629, which shall be closed with a commit fixing this.

Reply:
Since you highlighted me, I'll throw in my opinion about this...

I don't know where you see a mistreatment of the integer type here. The relation between 1/0 and true/false in C is just implicit. If `== true` was used, I'd agree with you.

`unit_movepos()` returns 0 on error and 1 on success, so checking for `== 1` is absolutely valid.
If custom code introduces a new return value, the one who adds it is responsible for validating all dependencies.

> There needs to be a compile error then.
How would you do this?

\**********
**But in the end, you are right. If `!= 0` is used, all possible side effects are excluded.** :+1:
\**********

Reply:
Ah, true, i totally agree.. I didn't thought about custom code when I did the change, sorry...
I may open a PR fixing it later today (~12 hours from now), but if you guys want to ship a follow up release earlier, feel free to do it before me. Thanks!

Reply:
With mistreatment of integer type as bool I simply refer to implicit comparisons such as `if (some_int)` which makes you think that it's checking for `== true`.

Seems like I was really sleepy yesterday and saw a return 2 where there wasn't any. So there are at least no side-effects in our code-base, not that I know of any. 😅 
But yeah let's use `== 0` and `!= 0` for stuff like this when just fixing up implict comparison on non-bool types. Essentially chaging those comparisons and only looking at specific return values should be done in a separate commit at least, so that it's more noticeable.